spec/lib/payment_calculator/ecf/contract/uplift_payment_calculations_spec.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require "payment_calculator/ecf/contract/uplift_payment_calculations"
+
+class DummyClass
+  include PaymentCalculator::Ecf::Contract::UpliftPaymentCalculations
+end
+
+describe ::PaymentCalculator::Ecf::Contract::UpliftPaymentCalculations do
+  it "returns the expected types for uplift" do
+    contract = double("Contract Double", uplift_amount: BigDecimal(100.00, 2))
+    call_off_contract = DummyClass.new({ contract: contract })
+    event_type = :started
+
+    if @combined_results.nil?
+      expect(call_off_contract.uplift_payment_per_participant.round(2)).to be_a(BigDecimal)
+      expect(call_off_contract.uplift_payment_per_participant_for_event(event_type: event_type)).to be_a(BigDecimal)
+      expect(call_off_contract.uplift_payment_for_event(event_type: event_type, uplift_participants: 100)).to be_a(BigDecimal)
+    end
+  end
+
+  it "performs calculations of the uplift payments" do
+    contract = double("Contract Double", uplift_amount: 100)
+    call_off_contract = DummyClass.new({ contract: contract })
+    total_participants = 330
+
+    expect(call_off_contract.uplift_payment_per_participant.round(0)).to eq(100.00)
+
+    %i[started].each do |event_type|
+      expect(call_off_contract.uplift_payment_per_participant_for_event(event_type: event_type).round(2)).to eq(100.00)
```

Comment:
I'd probably put the type checks for a single next to the value checks....
... like this:

```suggestion
      expect(call_off_contract.uplift_payment_per_participant_for_event(event_type: event_type).round(2)).to eq(100.00)
      expect(call_off_contract.uplift_payment_per_participant).to be_a(BigDecimal)
```

Reply:
That type is only applicable for the first one, not for every event type